pkg/controller/clusterdeployment/clusterdeployment_controller_test.go

```diff
@@ -208,10 +208,26 @@ func TestClusterDeploymentReconcile(t *testing.T) {
 				testCompletedInstallJob(),
 				testMetadataConfigMap(),
 				testSecret(corev1.SecretTypeOpaque, adminKubeconfigSecret, "kubeconfig", adminKubeconfig),
-				testSecret(corev1.SecretTypeDockercfg, pullSecretSecret, corev1.DockerConfigKey, "{}"),
+				testSecret(corev1.SecretTypeDockercfg, pullSecretSecret, corev1.DockerConfigJsonKey, "{}"),
 				testSecret(corev1.SecretTypeOpaque, sshKeySecret, adminSSHKeySecretKey, "fakesshkey"),
 			},
 		},
+		{
```

Comment:
We should also add tests to verify (1) that a merged pull secret gets created, (2) that a merged pull secret gets replaced when there is a change, and (3) that a merged pull secret does not get replaced when there is no change.